### PR TITLE
Replace the placeholder release build with our real one

### PR DESCRIPTION
**Pull request by DHowett**
This pull request ports our old release pipeline from Azure DevOps' editor to real YAML.

It includes the following changes on top of a straight-up "export" from Azure:

- Converts all queue-time variables into form-based parameters
- Adds a "matrix" build strategy for Configs * Platforms
- Renames all jobs to have reasonable names
- The YAML generator has a bug where it inlines scripts *and* file paths if a task had both; remove old inlines
- Removes dead rules
- Fixes the WPF build to include the apiset impostor
- Migrates the access token into the environment for the one build stage that needs it
- Cleans up some of the online script logic
- Removes all of the "!is pull request?" checks

### CODE REVIEW

**Comment by DHowett** [comment]
The new build execution UI is very nice:

image

**Review by carlos-zamora — APPROVED** [review]

File: build/pipelines/release.yml [hunk 1]
```diff
@@ -1,48 +1,483 @@
...
+
+        src\cascadia\TerminalConnection\Resources\en-US\Resources.resw
+
+        src\cascadia\TerminalSettingsModel\Resources\en-US\Resources.resw
+
+        src\cascadia\TerminalSettingsEditor\Resources\en-US\Resources.resw
+
+        src\cascadia\WindowsTerminalUniversal\Resources\en-US\Resources.resw
+
+        src\cascadia\CascadiaPackage\Resources\en-US\Resources.resw
```
[the remark this answers is not in the document]
Comment (DHowett) [reply build/pipelines/release.yml:125]: no.

[the remark this answers is not in the document]
Comment (DHowett) [reply build/pipelines/release.yml:125]: we probably can, but i'd rather not mess with steps that are working

File: build/pipelines/release.yml [hunk 2]
```diff
@@ -1,48 +1,492 @@
...
-#     <none>  - Development build resources (default)
-#     Preview - Preview build resources
-#     Release - regular build resources
+  TerminalInternalPackageVersion: "0.0.7"
+
+name: $(BuildDefinitionName)_$(date:yyMM).$(date:dd)$(rev:rrr)
+resources:
+  repositories:
+  - repository: self
+    type: git
```
Comment (miniksa) [inline build/pipelines/release.yml:55]: We should log a follow up on shallow checkout...

File: build/pipelines/release.yml [hunk 3]
```diff
@@ -1,48 +1,492 @@
...
+        src\cascadia\TerminalSettingsEditor\Resources\en-US\Resources.resw
+
+        src\cascadia\WindowsTerminalUniversal\Resources\en-US\Resources.resw
+
+        src\cascadia\CascadiaPackage\Resources\en-US\Resources.resw
+      appendRelativeDir: true
+      localizationTarget: false
+      pseudoSetting: Included
+  - task: PowerShell@2
+    displayName: Move Loc files one level up
```
Comment (miniksa) [inline build/pipelines/release.yml:130]: RIP in pieces to the previous description

File: build/pipelines/release.yml [hunk 4]
```diff
@@ -1,48 +1,492 @@
...
+      inputs:
+        artifactName: wpf-dll-x64-$(BuildConfiguration)
+        itemPattern: '**/*.dll'
+        downloadPath: bin\x64\$(BuildConfiguration)\
+        extractTars: false
+    - task: PowerShell@2
+      displayName: Move downloaded artifacts up a level
+      inputs:
+        targetType: inline
+        # Find all artifact files and move them up a directory. Ugh.
```
Comment (miniksa) [inline build/pipelines/release.yml:381]: Or was it this guy for the RIP in pepperoni. Either way. I'm still laughing/crying

**Review by miniksa — APPROVED** [review]

File: build/pipelines/release.yml (hunk 4 above)
Comment (DHowett) [reply build/pipelines/release.yml:381]: Haha I think it was this one

File: build/pipelines/release.yml (hunk 2 above)
Comment (DHowett) [reply build/pipelines/release.yml:55]: Done!

**Pull request merged**